AdobeStockImage/Test/Integration/Model/SaveImageTest.php

```diff
@@ -0,0 +1,213 @@
+                    'width' => 110,
+                    'title' => 'test',
+                    'content_type' => 'image/png',
+                    'height' => 210,
+                    'some_bool_param' => false,
+                    'some_nullable_param' => null,
+                    'extension_attributes' => [
+                        'title' => 'test',
+                        'is_downloaded' => 0,
+                        'is_licensed_locally' => 0,
+                        'thumbnail_240_url' => 'https://test.url/magento-logo.png',
+                        'creator_id' => 1122,
+                        'creator_name' => 'Test',
+                        'path' => 'catalog/category/tmp.png',
+                        'content_type' => 'image/png',
+                        'category' => [
+                            'id' => 1,
+                            'name' => 'Test'
+                        ],
+                    ]
+                ],
+                'sourcePath' => 'magento-logo.png',
+                'destinationPath' => 'catalog/category/tmp.png',
+            ]
+        ];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp(): void
+    {
+        $this->driver = Bootstrap::getObjectManager()->get(DriverInterface::class);
+        $this->fileSystem = Bootstrap::getObjectManager()->get(Filesystem::class);
+        $this->assetRepository = Bootstrap::getObjectManager()->get(AssetRepositoryInterface::class);
+        $this->criteriaBuilder = Bootstrap::getObjectManager()->get(SearchCriteriaBuilder::class);
+        Bootstrap::getObjectManager()->configure([
+            'preferences' => [
+                Https::class => HttpsDriverMock::class
+            ]
+        ]);
+        $this->saveImage = Bootstrap::getObjectManager()->create(SaveImageInterface::class);
+    }
+
+    /**
+     * Test with image.
+     *
+     * @param array $documentData
+     * @param string $sourceFile
+     * @param string $destinationPath
+     * @return void
+     * @dataProvider getSaveTestDataProvider
+     */
+    public function testSave(array $documentData, string $sourceFile, string $destinationPath): void
+    {
+        $this->deleteImage($destinationPath);
+        $document = $this->getDocument($documentData);
+        $mediaDir = $this->fileSystem->getDirectoryWrite(DirectoryList::MEDIA);
+        $this->saveImage->execute(
+            $document,
+            $this->getImageFilePath($sourceFile),
+            $mediaDir->getAbsolutePath($destinationPath)
+        );
+        self::assertTrue(
+            $this->fileSystem->getDirectoryRead(DirectoryList::MEDIA)->isExist($destinationPath),
+            'File was not saved by destination'
+        );
+        $searchCriteria = $this->criteriaBuilder
+            ->addFilter('creator_id', $document->getCustomAttribute('creator_id')->getValue(), 'eq')
+            ->create();
+        self::assertNotEmpty(
```

Comment:
Please add a bit more assertions, check the asset data to ensure the loaded asset represents the data passed to the saveImage service